File: netlify/functions/get_categories.py
import json
import os
from pymongo import MongoClient

# MongoDB connection - using standard environment variable names
MONGODB_URI = os.environ.get('MONGODB_URI', 'mongodb+srv://root:root12345@cluster0.mongodb.net/ai_tools_db?retryWrites=true&w=majority')
client = None

def get_db_client():
    global client
    if client is None:
        client = MongoClient(MONGODB_URI)
    return client

def lambda_handler(event, context):
    try:
        # Handle CORS preflight
        if event.get('httpMethod') == 'OPTIONS':
            return {
                'statusCode': 200,
                'headers': {
                    'Access-Control-Allow-Origin': '*',
                    'Access-Control-Allow-Headers': 'Content-Type',
                    'Access-Control-Allow-Methods': 'GET, OPTIONS'
                },
                'body': ''
            }

        # Get database client
        client = get_db_client()
        db = client.ai_tools_db
        collection = db.tools
        
        # Get distinct categories
        categories = collection.distinct('category')
        
        # Sort categories alphabetically
        categories.sort()
        
        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Headers': 'Content-Type',
                'Access-Control-Allow-Methods': 'GET, OPTIONS'
            },
            'body': json.dumps({
                'categories': categories
            })
        }
        
    except Exception as e:
        print(f"Error in get_categories: {str(e)}")
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'error': 'Internal server error',
                'detail': str(e)
            })
        }

File: netlify/functions/get_categories.js
const { MongoClient } = require("mongodb")

// MongoDB connection - established outside handler for reuse
const uri =
  process.env.MONGODB_URI || "mongodb+srv://root:root12345@cluster0.mongodb.net/ai_tools_db?retryWrites=true&w=majority"
const client = new MongoClient(uri)
const clientPromise = client.connect()

exports.handler = async (event, context) => {
  try {
    // Handle CORS preflight
    if (event.httpMethod === "OPTIONS") {
      return {
        statusCode: 200,
        headers: {
          "Access-Control-Allow-Origin": "*",
          "Access-Control-Allow-Headers": "Content-Type",
          "Access-Control-Allow-Methods": "GET, OPTIONS",
        },
        body: "",
      }
    }

    // Connect to database
    const client = await clientPromise
    const db = client.db("ai_tools_db")
    const collection = db.collection("tools")

    // Get distinct categories
    const categories = await collection.distinct("category")

    // Sort categories alphabetically
    categories.sort()

    return {
      statusCode: 200,
      headers: {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Headers": "Content-Type",
        "Access-Control-Allow-Methods": "GET, OPTIONS",
      },
      body: JSON.stringify({
        categories: categories,
      }),
    }
  } catch (error) {
    console.error("Error in get_categories:", error)
    return {
      statusCode: 500,
      headers: {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
      },
      body: JSON.stringify({
        error: "Internal server error",
        detail: error.message,
      }),
    }
  }
}
